testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/GradleExecuter.java

```diff
@@ -402,6 +402,16 @@ default GradleExecuter expectDeprecationWarningWithMultilinePattern(String patte
      */
     GradleExecuter noDeprecationChecks();
 
+    /**
+     * Disable automatic Java version deprecation warning checks.
+     * <p>
+     * By default, the executor will ignore all deprecation warnings related to running a build
+     * on a Java version that will no longer be supported in future versions of Gradle. In most
+     * cases we do not care about this warning, but when we want to explicitly test that a build
+     * does emit this warning, we disable this check.
+     */
+    GradleExecuter noJavaVersionDeprecationChecks();
```

Comment:
❓ I think this method is named / described incorrectly. `noDeprecationChecks` stops a build from failing when a deprecation is present. This method _starts_ failing a build when this specific deprecation is present. I think this should be `disableDaemonJavaVersionDeprecationFiltering()`, as that's what it's controlling.